src/D2L.Security.OAuth2/Keys/Default/JsonWebKeySet.cs

```diff
@@ -17,13 +17,15 @@ internal sealed class JsonWebKeySet {
 			Source = src ?? throw new ArgumentNullException( nameof( src ) );
 
 			try {
-				var data = JsonSerializer.Deserialize<Dictionary<string, List<object>>>( json );
+				var data = JsonSerializer.Deserialize<Dictionary<string, object>>( json );
 
 				if( !data.ContainsKey( "keys" ) ) {
 					throw new JsonWebKeyParseException( "invalid json web key set: missing keys array" );
 				}
 
-				List<object> keyObjects = data["keys"];
+				if ( data["keys"] is not IEnumerable<object> keyObjects ) {
+					throw new JsonWebKeyParseException( "invalid json web key set: keys not an array" );

```

Comment:
I have no idea why tests are failing CI because of this. They're passing locally. Any ideas what I should try?

Reply:
Thanks. Looks like a .net 6 issue where it's a JsonElement.